src/PlaywrightSharp/FrameManager.cs

```diff
@@ -120,6 +120,22 @@ internal void FrameLifecycleEvent(string frameId, string name)
             }
         }
 
+        internal void FrameCommittedSameDocumentNavigation(string frameId, string url)
+        {
+            if (!Frames.TryGetValue(frameId, out var frame))
+            {
+                return;
+            }
+
+            frame.Url = url;
+            foreach (var watcher in LifecycleWatchers)

```

Comment:
this could throw if another event is dispatched during the iteration adding another watcher to `LifecycleWatchers`. you might need to call `ToArray`